### PR TITLE
Upgrade gardener/gardener to v1.80,k8s.io/* to v0.28, sigs.k8s.io/controller-runtime to v0.16

### PR DESCRIPTION
**Pull request by ary1992**
**How to categorize this PR?**

/area open-source
/kind enhancement
/platform alicloud

**What this PR does / why we need it**:
Vendor `gardener/gardener` to `v1.80.3`.
Upgrade `k8s.io/*` to `v0.27.5`
Upgrade `sigs.k8s.io/controller-runtime` to `v0.16.2`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Golang verison is upgraded in `branch.cfg` to `1.21.1` in commit https://github.com/gardener/gardener-extension-provider-alicloud/commits/930072fa

**Release note**:

```other operator
The following golang dependencies have been upgraded :
- `gardener/gardener` to `v1.80.3`
- `k8s.io/*` to `v0.27.5`
- `sigs.k8s.io/controller-runtime` to `v0.16.2`
```

### CODE REVIEW

**Comment by shaoyongfeng** [comment]
Hi @ary1992 ,
can you rebase your PR and make sure all checks passed?

**Comment by shafeeqes** [comment]
@shaoyongfeng PTAL!

File: .test-defs/bastion-test.yaml [hunk 1]
```diff
@@ -16,4 +16,4 @@ spec:
       --access-key-secret=$ACCESS_KEY_SECRET
       --region=$REGION
 
-  image: eu.gcr.io/gardener-project/3rd/golang:1.19.3
+  image: eu.gcr.io/gardener-project/3rd/golang:1.21.1
```
Comment (shaoyongfeng) [inline .test-defs/bastion-test.yaml:19]: this image does't exist. you may need to use golang:1.21.1

**Review by shaoyongfeng — APPROVED** [review]

**Pull request merged**